core/cache/application.go

```diff
@@ -28,18 +33,43 @@ type Application struct {
 	configHash string
 }
 
-func (m *Application) setDetails(details ApplicationChange) {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	m.details = details
+func (a *Application) WatchFields(comparitors ...func(*ApplicationDelta) bool) *ApplicationFieldWatcher {
+	w := newApplicationFieldWatcher(comparitors)
+
+	unsub := a.hub.Subscribe(a.topic(applicationFieldChange), w.detailsChange)
```

Comment:
This is going to be valid only if we have a simple hub per model. Has that been done yet?

Reply:
Good point. For now, I have prefixed the application topic with model UUID.